src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlNodeReader.cs

```diff
@@ -1018,7 +1018,7 @@ internal string LookupPrefix(string namespaceName)
 
             if (scope != XmlNamespaceScope.Local)
             {
-                if (dict.ContainsKey(string.Empty) && dict[string.Empty] == string.Empty)
+                if (dict.ContainsKey(string.Empty) && dict[string.Empty].Length == 0)
```

Comment:
What guarantees the value in the dictionary here is non-null?

Reply:
Same as previous:

> Well, I think nothing guarantees that.
> Should it be string.IsNullOrEmpty then? or reverted back?
> I don't know what should happen if it was null.

Reply:
If it's null, your change would cause it to throw an exception where it wouldn't have before.  The benefit of the change being made here is not worth the risk.  All such questionable cases should be reverted.  Thanks.

Reply:
```suggestion
                if (dict.ContainsKey(string.Empty) && dict[string.Empty] == string.Empty)
```